### PR TITLE
Fixed discovery services based on 'ThingHandlerService'

### PR DESCRIPTION
**Pull request by cweitkamp**
- Fixed discovery services based on 'ThingHandlerService'

See https://github.com/openhab/openhab-core/pull/709#issuecomment-481726321

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/discovery/DigiplexDiscoveryService.java [hunk 1]
```diff
@@ -137,21 +131,25 @@ public void handleAreaLabelResponse(@NonNull AreaLabelResponse response) {
...
     }
 
+    @Override
+    public void activate() {
+        super.activate(null);
+    }
+
     @Override
     public void deactivate() {
+        removeOlderResults(getTimestampOfLastScan());
```
Comment (martinvw) [inline bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/discovery/DigiplexDiscoveryService.java]: Do we want this to happen?

**Comment by martinvw** [comment]
The rest of changes look good to me
FTR: the build error seems to be caused by the Log Tailer

File: bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/discovery/DigiplexDiscoveryService.java (hunk 1 above)
Comment (cweitkamp) [reply bundles/org.openhab.binding.digiplex/src/main/java/org/openhab/binding/digiplex/internal/discovery/DigiplexDiscoveryService.java]: No, we do not. I can remember. But there are tons of bindings doing it at the time the service will be deactivated. I will change it for the touched bindings in this PR.

**Comment by martinvw** [comment]
Lets wait for the build

**Comment by davidgraeff** [comment]
Travis agrees

**Pull request merged**